### DISCUSSION TITLE
NodeJS/NPM auto-download

### DISCUSSION THREAD

**Discussion by ia3andy** [quarkiverse/quarkus-quinoa#114] [Polls]

**Comment by maxandersen** [comment]
which version/distro of npm/nodejs would it download?

**Reply by ia3andy** [reply]
I think I would use https://mvnrepository.com/artifact/com.github.eirslett/frontend-plugin-core, it already contains most of the code:
https://github.com/eirslett/frontend-maven-plugin/blob/master/frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NodeInstaller.java

**Reply by ia3andy** [reply]
Another option would be to use docker, but I am not sure about perf, this needs to be quick..

**Comment by ia3andy** [comment]
This is implemented in #173.

Supporting yarn should be fairly easy if anyone is interested in implementing it.

**Reply by melloware** [reply]
Yarn is now implemented, marking this as answered
